autoscheduler/frontend/src/components/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/ExpandedCourseCard.tsx

```diff
@@ -79,15 +94,27 @@ const ExpandedCourseCard: React.FC<ExpandedCourseCardProps> = ({
       </div>
       <div className={styles.content}>
         <Autocomplete
-          options={['CSCE 121', 'MATH 151', 'CSCE 221']}
+          options={options}
           size="small"
+          autoHighlight
+          autoSelect
+          inputValue={inputValue}
           value={course}
-          onChange={(evt: React.ChangeEvent, val: string): void => {
+          multiple={false}
+          onClose={(): void => {
+            if (!options.find((val) => val === inputValue)) setInputValue('');
+          }}
+          onChange={(evt: object, val: string): void => {
             dispatch(updateCourseCard(id, {
               course: val,
-            }));
+            }, term));
+          }}
+          onInputChange={(evt: object, val: string, reason: string): void => {
+            setInputValue(val);
+            if (val === '') return; // Skip empty inputs
```

Comment:
Can you `setOptions([])` before returning here? Because right now, it leaves the suggestions list after you clear the input, which feels a little off. Other than that, this PR looks good to me.